shoes-core/lib/shoes/widget.rb

```diff
@@ -21,40 +26,47 @@ class Shoes
   #     say_hello 'Shoes'
   #   end
   #
-  class Widget
-    include Common::Inspect
-
+  class Widget < Shoes::Flow
     Shoes::App.subscribe_to_dsl_methods self
 
-    attr_accessor :parent
-    attr_writer :app
+    attr_accessor :original_args
 
-    class << self
-      attr_accessor :app
+    # Having Widget define initialize makes it easier for us to detect whether
+    # subclasses have inappropriately overridden it or not.
+    def initialize(*_)
+      super
     end
 
-    # lookup a bit more complicated as during initialize we do
-    # not have @app yet...
-    def app
-      @app || self.class.app
+    def initialize_widget
```

Comment:
shouldn't this be `(*args)` ?

Reply:
Actually I don't think so. If we made this `*args`, then widgets that don't define an `initialize_widget` would happily take any number of args to the widget method we generate, but then not do anything with them. Seems better that we accept none by default, and would give an argument count error if you tried to mis-call it. That make sense?

Reply:
makes sense yeah :)